### PR TITLE
Feat/condo/sberdoma 62/apply createapp

### PR DESCRIPTION
**Pull request by pahaz**

### CODE REVIEW

File: apps/condo/pages/ticket/index.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import qs from 'qs'
 import { PageContent, PageHeader, PageWrapper } from '../../containers/BaseLayout'
 import { OrganizationRequired } from '../../containers/OrganizationRequired'
 
-import { Ticket } from '../../utils/clientSchema/Ticket'
+import { Ticket } from '@condo/domains/ticket/utils/clientSchema'
```
Comment (pahaz) [inline apps/condo/pages/ticket/index.tsx:13]: ! it's just one place!

File: apps/condo/domains/ticket/utils/clientSchema/Ticket.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    status?: string
+    source?: string
+    classifier?: string
+    assignee?: string
+    operator?: string
+    client?: string
+}
+
+function convertToUIFormState (state: ITicketUIState): ITicketUIFormState {
+    if (!state) return {} as ITicketUIFormState
```
Comment (Dimitreee) [inline apps/condo/domains/ticket/utils/clientSchema/Ticket.ts:37]: We shouldn't use compile-time casts. Better try to describe more understandable function signature and return type.

Comment (Dimitreee) [inline apps/condo/domains/ticket/utils/clientSchema/Ticket.ts:37]:
```suggestion
function convertToUIFormState (state: ITicketUIState): ITicketUIFormState|undefined {
    if (!state) return
```

File: packages/@core.codegen/templates/app00/index.js [hunk 3]
```diff
@@ -18,21 +18,17 @@ const keystone = new Keystone({
...
-                        keystone,
-                        listKey,
-                        items,
-                    })
-                }
+        const users = await keystone.lists.User.adapter.findAll()
+        if (!users.length) {
+            const initialData = require('./initialData')
+            for (let { listKey, items } of initialData) {
+                console.log(`🗿 createItems(${listKey}) -> ${items.length}`)
```
Comment (Dimitreee) [inline packages/@core.codegen/templates/app00/index.js:25]: 🗿)))))

File: apps/condo/README.md [hunk 4]
```diff
@@ -47,4 +47,33 @@ docker-compose run app yarn workspace @app/condo makemigrations
 
 # migrate current DB to new schema
 docker-compose run app yarn workspace @app/condo migrate
+
+# if you needed to rollback applyed migration
```
Comment (pahaz) [inline apps/condo/README.md:51]:
```suggestion
# if you want to rollback applied migration
```

File: apps/condo/domains/common/utils/codegeneration/generate.hooks.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+import { ApolloError } from '@apollo/client'
+
+import { useMutation, useQuery } from '@core/next/apollo'
+import { useIntl } from '@core/next/intl'
+
+const getObjects = (objectsContainer, converter) => (objectsContainer && objectsContainer.objs) ? objectsContainer.objs.map(converter) : []
+
+interface IHookConverters<GQL, GQLInput, UI, UIForm> {
+    convertToGQLInput: (state: UIForm) => GQLInput
+    convertToUIState: (x: GQL) => UI
```
Comment (pahaz) [inline apps/condo/domains/common/utils/codegeneration/generate.hooks.ts:11]:
```suggestion
    convertToUIState: (item: GQL) => UI
```

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 6]
```diff
@@ -28,55 +26,61 @@ const Ticket = new GQLListSchema('Ticket', {
...
             isRequired: true,
             defaultValue: 0,
-            schemaDoc: 'Counter showing the number of changes `status` to `new_or_reopened`. Is not auto changed',
             access: {
                 read: true,
                 update: false,
                 create: false,
             },
         },
+        // TODO(Dimitreee): server side autogen
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:41]:
```suggestion
        // TODO(Dimitreee): server side auto generation
```

File: apps/condo/domains/ticket/utils/clientSchema/Ticket.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+const RELATIONS = ['status', 'client', 'operator', 'assignee', 'classifier', 'organization', 'source', 'property', 'executor', 'related']
+
+interface ITicketUIState extends Ticket {
+    id: string
+    organization: Organization
+}
+
+function convertToUIState (item: Ticket): ITicketUIState {
+    if (item.dv !== 1) throw new Error('unsupported item.dv')
+    return pick(item, FIELDS) as ITicketUIState
```
[the remark this answers is not in the document]
Comment (pahaz) [reply apps/condo/domains/ticket/utils/clientSchema/Ticket.ts:23]: It's hard for TS to understand the return type.

File: apps/condo/domains/ticket/utils/clientSchema/Ticket.ts (hunk 2 above)
Comment (pahaz) [reply apps/condo/domains/ticket/utils/clientSchema/Ticket.ts:37]: If you remove this line, there will be problems with auto-completion and the current pages will fall. 
To be honest, you added this line, I tried to remove it, but it breaks the front.

Comment (pahaz) [reply apps/condo/domains/ticket/utils/clientSchema/Ticket.ts:37]: the current pages will fall

**Review by Dimitreee — APPROVED** [review]

**Pull request merged**